### PR TITLE
Implement S3 directories in set_dir

### PR DESCRIPTION
**Pull request by dimaryaz**

### CODE REVIEW

File: api/python/t4/packages.py [hunk 1]
```diff
@@ -461,10 +464,21 @@ def set_dir(self, lkey, path):
...
                 logical_key = lkey + f.relative_to(src_path).as_posix()
                 # TODO: Warn if overwritting a logical key?
                 self.set(logical_key, entry)
+        elif url.scheme == 's3':
+            src_bucket, src_key, src_version = parse_s3_url(url)
+            if src_version:
+                raise PackageException("Directories cannot have versions")
+            objects = list_objects(src_bucket, src_key)
+            for obj in objects:
+                obj_url = 's3://%s/%s' % (src_bucket, quote(obj['Key']))
```
Comment (meffij) [inline api/python/t4/packages.py]: what about version ID?

File: api/python/t4/packages.py [hunk 2]
```diff
@@ -461,10 +464,21 @@ def set_dir(self, lkey, path):
...
+                entry = PackageEntry([obj_url], None, None, None)
+                logical_key = lkey + obj['Key'][len(src_key) + 1:]
```
Comment (meffij) [inline api/python/t4/packages.py]: Is there any way to make this line more clear? I'm not sure why you're doing it that way

File: api/python/t4/packages.py (hunk 1 above)
Comment (dimaryaz) [reply api/python/t4/packages.py]: Ooh, yes

File: api/python/t4/packages.py (hunk 2 above)
Comment (dimaryaz) [reply api/python/t4/packages.py]: Don't know if there is a way - we have to strip the prefix.

E.g., if I call `set_dir('foo', 'x/y')`, S3 will give me keys like:
```
x/y/a.txt
x/y/b.txt
```
and I need to turn them into:
```
foo/a.txt
foo/b.txt
```
So not much to do other than substring?

**Pull request merged**